testutil/blobfactory/payforblob_factory.go

```diff
@@ -0,0 +1,268 @@
+package blobfactory
```

Comment:
we could probably still use a refactor our to our testing utilities, but these I think this is a step in the right direction.